Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -750,8 +754,25 @@ class BrowserViewController: UIViewController {
     }
     
     func updateTabsBarVisibility() {
+        func shouldShowTabBar() -> Bool {
+            let tabCount = UIApplication.isInPrivateMode ? tabManager.privateTabs.count : tabManager.normalTabs.count
```

Comment:
We should remove the concept of `privateTabs` and `normalTabs` to just `tabs`. Let the result be returned solely based on the current mode. This may be too large of a refactor for now, since it completely breaks the TabManager's concept of private tab (e.g. no longer utilizing coredata).

Reply:
Also I think using `isEmpty` is probably better than using all of the `> 1` checks later down. Just make `tabCount` a boolean `hasTabs` or something.

Reply:
I can dump the use if private vs normal tabs, but isEmpty doesn't check for > 1, it checks for > 0, so it would show the tab bar even if there's only 1 tab. If thats cool, I'll do that.